sympy/utilities/lambdify.py

```diff
@@ -422,6 +424,9 @@ def lambdify(args, expr, modules=None, printer=None, use_imps=True,
                 names.append('arg_' + str(n))
 
     # Create lambda function.
+    if not should_dummify(expr, printer=printer):
+        dummify = False
```

Comment:
If you move it to lambdastr, I would remove the should_dummify function and add the relevant code directly to lambdastr.